sdk/identity/azure-identity/src/main/java/com/azure/identity/SharedTokenCacheCredential.java

```diff
@@ -41,31 +43,44 @@
      * @param clientId the client ID of the application
      * @param identityClientOptions the options for configuring the identity client
      */
-    SharedTokenCacheCredential(String username, String clientId, IdentityClientOptions identityClientOptions) {
+    SharedTokenCacheCredential(String username, String clientId, String tenantId,
+                               IdentityClientOptions identityClientOptions) {
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
 
         if (username == null) {
             this.username = configuration.get(Configuration.PROPERTY_AZURE_USERNAME);
         } else {
             this.username = username;
         }
-
-        this.clientId = clientId;
+        if (clientId == null) {
+            this.clientId = configuration.get(Configuration.PROPERTY_AZURE_CLIENT_ID);
```

Comment:
We're giving priority to env config over the values passed through the builder ?